api/commands/run.ts

```diff
@@ -74,7 +74,7 @@ export class Runner {
         );
 
         // Start the main execution loop.
-        this.loop(() => resolve(this.result), reject);
+        let runLoop = this.loop(() => resolve(this.result), reject);
```

Comment:
this is actually unsafe: we should not add a dummy variable, but instead add a .catch handler for any thrown Errors.